src/praisonai-agents/test_agent_guardrails.py

```
@@ -0,0 +1,63 @@
+            return True, task_output
+        
+        # Test Agent creation with function guardrail
+        agent1 = Agent(
+            name="TestAgent1",
+            instructions="You are a test agent",
+            guardrail=test_guardrail,
+            max_guardrail_retries=2
+        )
+        print("✓ Agent with function guardrail created successfully")
+        print(f"  - Agent name: {agent1.name}")
+        print(f"  - Has guardrail function: {agent1._guardrail_fn is not None}")
```

Pylint (3.3.7)

[warning] Access to a protected member _guardrail_fn of a client class

(W0212)